include/openthread/border_router.h

```diff
@@ -57,20 +57,29 @@ extern "C" {
  *
  * @note  This method MUST be called before any other otBorderRouting* APIs.
  *
- * @param[in]  aInstance      A pointer to an OpenThread instance.
- * @param[in]  aInfraIfIndex  The infrastructure interface index.
+ * @param[in]  aInstance                 A pointer to an OpenThread instance.
+ * @param[in]  aInfraIfIndex             The infrastructure interface index.
+ * @param[in]  aInfraIfIsRunning         A boolean that indicates whether the infrastructure
+ *                                       interface is running.
+ * @param[in]  aInfraIfLinkLocalAddress  A pointer to the IPv6 link-local address of the infrastructure
+ *                                       interface. NULL if the IPv6 link-local address is missing.
  *
  * @retval  OT_ERROR_NONE          Successfully started the Border Routing manager on given infrastructure.
- * @retval  OT_ERROR_INVALID_ARGS  The index of the infra interface is not valid.
+ * @retval  OT_ERROR_INVALID_ARGS  The index or the IPv6 link-local address of the infra interface is not valid.
  * @retval  OT_ERROR_FAILED        Internal failure. This is usually failed to generate random prefixes.
  *
+ * @sa otPlatInfraIfStateChanged.
+ *
  */
-otError otBorderRoutingInit(otInstance *aInstance, uint32_t aInfraIfIndex);
+otError otBorderRoutingInit(otInstance *        aInstance,
```

Comment:
It seems this API is not necessary, and all the information can be delivered by the otPlatInfraIfStateChanged() API. I'm fine to deal with this issue in separate PRs.